app/src/main/java/com/infomaniak/mail/utils/UiUtils.kt

```diff
@@ -130,4 +133,61 @@ object UiUtils {
     }
 
     fun dividerDrawable(context: Context) = AppCompatResources.getDrawable(context, R.drawable.divider)
+
+    /**
+     * Only returns a quantity of at most 2, used to differentiate between the singular or plural form of the dialog messages
+     */
+    fun findExternalRecipientsInThread(
+        thread: Thread,
+        emailDictionary: MergedContactDictionary,
+        aliases: List<String>,
+    ): Pair<String?, Int> {
+        var externalRecipientEmail: String? = null
+        var externalRecipientQuantity = 0
+
+        thread.messages.forEach { message ->
+            val (singleEmail, quantityForThisMessage) = findExternalRecipientInIterables(emailDictionary, aliases, message.from)
+
+            externalRecipientQuantity += quantityForThisMessage
+            if (externalRecipientQuantity > 1) return null to 2
+
+            if (quantityForThisMessage == 1) externalRecipientEmail = singleEmail
+        }
+
+        return externalRecipientEmail to externalRecipientQuantity
+    }
+
+    fun findExternalRecipientInDraft(
+        draft: Draft,
+        aliases: List<String>,
+        emailDictionary: MergedContactDictionary,
+    ): Pair<String?, Int> {
+        return findExternalRecipientInIterables(emailDictionary, aliases, draft.to, draft.cc, draft.bcc)
+    }
+
+    /**
+     * Only returns a quantity of at most 2, used to differentiate between the singular or plural form of the dialog messages
+     */
+    private fun findExternalRecipientInIterables(
+        emailDictionary: MergedContactDictionary,
+        aliases: List<String>,
+        vararg recipientLists: Iterable<Recipient>,
+    ): Pair<String?, Int> {
```

Comment:
What about putting these 3 functions as Extension of Thread, Draft and Iterable<Recipient> ?

Reply:
Even better was what we talked about with Joris but ended up forgetting apparently. I extracted them to their own util